### DISCUSSION TITLE
GitHub Actions stability + billing incident

### DISCUSSION THREAD

**Discussion by DamienDennehy** [community/community#14900] [Actions]
First of all, #HugOps. Rough week.

The March 2022 [Availability Report](https://github.blog/2022-04-06-github-availability-report-march-2022/) was a good read to understand several issues that happened last month. Unfortunately issues seem to be still occurring.

Looking at the [Incident Reports](https://www.githubstatus.com/history)...

- In February there were 3 GitHub Actions issues.
- In March there were 9 GitHub Actions issues.
- In April there have been 2 so far. 3 counting today's billing incident.

Today my company was billed $24K for Actions use when last month's bill was less than $100. There are other people on Twitter that have received billing emails of $120 million. I'd love to have seen the accounting department reactions when they saw that email 🤣

Because I had a spending limit on our account it means that until the billing issue is resolved, we are blocked from running Actions that use uploads for storage and or packages. (EDIT: This was resolved and we could work away while the billing issue was being worked on).

I'm sure these issues will be resolved, but as someone who wanted to move his entire team to GitHub Actions by the end of the year, it's certainly been a time to pause and reflect on this. We are very interested in seeing the April Engineering Report and some information on longer term plans for Actions stability would be really good.

**Comment by AntonioKoB** [comment]
I use the free tier of 2000 mins. And weirdly I noticed that the actions are counting (at least) double lately.
I did a test this morning. I had 1653 mins spent. The workflow took 5 minutes to run. I was expecting 1658 mins but the billing was showing 1663 (so doubled).
The workflow is a single job action.

I hope this gets fixed soon.

By the way. I usually don't consume more than 1000 - 1200 mins a month. This month weirdly I received an email warning I had reached 75% of the limit. I don't believe I did more development than usual this month.

**Comment by jledoux-sonergia** [comment]
I also noticed today that some of my workflow runs billable time (7m) were more than twice the total duration (3m15s)! 
Knowing that one of the jobs (about 1m) is executed on a self hosted runner, I'm really confused about the way the calculation is done... :open_mouth: 
Could it be than one job running for a few seconds gets billed for a minute?

**Reply by airtower-luna** [reply]
Yes, that's documented [here](https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions#minute-multipliers):

> GitHub rounds the minutes each job uses up to the nearest minute.

**Reply by jledoux-sonergia** [reply]
Thanks I've just realized that... that's a bit of a pain... I guess I'll have to optimize my jobs in regard to this
